teps/0064-tekton-cli-workspace-from-local-source.md

```diff
@@ -0,0 +1,95 @@
+---
+status: proposed
+title: Tekton CLI workspace from local source
+creation-date: '2021-05-06'
+last-updated: '2021-05-06'
+authors:
+- '@vdemeester'
+- '@sm43'
+---
+
+# TEP-0064: Tekton CLI workspace from local source
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Use Cases](#use-cases)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes a way to allow users to use local source code in the pipeline that executes in a kubernetes cluster.
+
+## Motivation
+
+Currently, users need to push their code to remote git repositories and use the references in
+pipeline. 
+
+This would simplify the development process for users as they could use their local source code
+directly in the pipeline without having to push to git repository.
+
+This will allows users to easily test their code in the pipeline.
+
+### Goals
+
+Provide ways to use local source code in the pipeline that executes in a kubernetes cluster.
+
+### Use Cases 
+
+- As a developer, I have a Tekton Pipeline running in a kubernetes cluster. If there are some modification locally and I want to see how the CI behaves prior pushing the code to a repository or before creating a pull request.
+
+## Proposal
+
+One of the way, this can be achieved is
+- User provides the path to the source as a directory which has to be copied   
+- Create a volume. As pvc is most commonly used, we create a pvc with workspace name defined in the pipeline. The type and storage capacity of pvc can have default values and user can overide them if required.
```

Comment:
Doesn't this normally mean if my team has a Tekton CI job which normally does a `git clone` or similar as a first task, I need to rewrite my CI pipeline to skip that stage and take the PVC which already has data in it? That seems like it pushes more work onto the user.